lib/graphql/batch/executor.rb

```diff
@@ -36,7 +36,7 @@ def tick
     end
 
     def wait_all
-      tick until loaders.empty?
+      tick until loaders.values.all? { |loader| loader.resolved? }
```

Comment:
It doesn't look like this is needed, since Loader#resolve returns early if it is already resolved.